### DISCUSSION TITLE
function to solve equations

### DISCUSSION THREAD

**Discussion by FWeinb** [josdejong/mathjs#2689] [Ideas]
I don't know if this is in the scope of this project but it would be great to have a solve function. 

Something like this:

```
x = solve(f = x + 1, x) 
```

where `x` will become:

```
x =  f - 1 
```

I will try to accomplish this by myself if I have to. 
Thanks for this awesome library!

**Comment by josdejong** [comment]
A symbolic equation solver would be great of course :). In the same category as symbolic integration and differentiation (#35). This kind of stuff is not so trivial though.

**Comment by DavidDoesDev** [comment]
Yeah, this would be a great add to an already great library! Adding that feature is a bit above my javascript pay grade—but for the person who can tackle this, check out this project: https://github.com/aaronpowell/mathy.js

I think this guy did a real nice implementation. Might be nice to emulate some of his architecture.
